client/src/components/Sub_RelatedItems/SingleProd.jsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+
+import ProdImg from './Sub_SingleProd/ProdImg.jsx';
+import ProdInfo from './Sub_SingleProd/ProdInfo.jsx';
+
+function SingleProd({ product, setProductID, productStyle, reviewMetadata, onClick }) {
+
+  // =================== STATES ===================
+
+  const [theseStyles, setTheseStyles] = useState([]);
+
+  const [thisPrice, setThisPrice] = useState([]);
+
+  const [thisAvgRating, setThisAvgRating] = useState([]);
+
+  const [imgUrl, setImgUrl] = useState('');
+
+  // =================== EFFECTS ===================
+
+  useEffect(() => {
+    axios.get(`/api/products/${id}/styles`)
+      .then((styles) => {
+        const newStyles = styles.data.results;
+        setTheseStyles(newStyles);
+        setImgUrl('');
+        setImgUrl(newStyles[0].photos[0].url);
+        return newStyles.filter(style => style['default?'] === true);
+      })
```

Comment:
What will happen if none of the styles have a default property set to true? I've tested multiple product IDs and found that some products don't have a valid default style, meaning that all styles have the default property set to false.

Reply:
There is a conditional in the `setPrice` helper that handles this edge case (same file, lines 52-64)